tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/OpenPaasContactsConsumerModule.java

```diff
@@ -0,0 +1,80 @@
+package com.linagora.tmail;
+
+import static com.linagora.tmail.OpenPaasModule.OPENPAAS_INJECTION_KEY;
+
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+import org.apache.james.backends.rabbitmq.RabbitMQConfiguration;
+import org.apache.james.backends.rabbitmq.RabbitMQConnectionFactory;
+import org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPool;
+import org.apache.james.backends.rabbitmq.SimpleConnectionPool;
+import org.apache.james.metrics.api.GaugeRegistry;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.utils.InitializationOperation;
+import org.apache.james.utils.InitilizationOperationBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.multibindings.ProvidesIntoSet;
+import com.linagora.tmail.configuration.OpenPaasConfiguration;
+import com.linagora.tmail.contact.OpenPaasContactsConsumer;
+import com.linagora.tmail.contact.OpenPaasContactsConsumerReconnectionHandler;
+
+public class OpenPaasContactsConsumerModule extends AbstractModule {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenPaasContactsConsumerModule.class);
+
+    @ProvidesIntoSet
+    public InitializationOperation initializeContactsConsumer(OpenPaasContactsConsumer instance) {
+        return InitilizationOperationBuilder
+            .forClass(OpenPaasContactsConsumer.class)
+            .init(instance::start);
+    }
+
+    @Provides
+    @Named(OPENPAAS_INJECTION_KEY)
+    @Singleton
+    public RabbitMQConfiguration provideRabbitMQConfiguration(RabbitMQConfiguration commonRabbitMQConfiguration,
+                                                              OpenPaasConfiguration openPaasConfiguration) {
+        Preconditions.checkArgument(openPaasConfiguration.contactConsumerConfiguration().isPresent(),
+            "OpenPaasConfiguration should have a contact consumer configuration");
```

Comment:
Changing `contactConsumerConfiguration` from mandatory to Optional (but indeed it is still mandatory and being `.get` everywhere) sounds "unsafe" to me. 

As the caller if I get the Optional `contactConsumerConfiguration`, I tend to understand that I should handle the case it could be empty. Reading the upstream code to see if it is always present seems not clear to me.

Reply:
> Changing contactConsumerConfiguration from mandatory to Optional (but indeed it is still mandatory and being .get everywhere) sounds "unsafe" to me.

not really everywhere, it only `.get` in the class related to "contact consumer" => then it true. 
The reason I changed it to optional, because when choosing cardDav, then "maybe" we don't need rabbitmq configured, but still require openpaas (base api url, admin, password)